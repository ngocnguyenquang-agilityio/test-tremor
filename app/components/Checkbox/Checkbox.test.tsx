// Libs
import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";

// Components
import CheckBox from "./Checkbox";

const handleCheckboxMock = jest.fn();

const CheckboxProps = {
  tabIndex: 3,
  checked: true,
  handleCheckBox: handleCheckboxMock,
};

const CheckboxComponent = () => render(<CheckBox {...CheckboxProps} />);

describe("CheckBox Component", () => {
  test("renders Checkbox component snapshot correctly", () => {
    const { container } = CheckboxComponent();

    expect(container).toMatchSnapshot();
  });

  test("renders CheckBox component with provided props", () => {
    const { getByTestId } = CheckboxComponent();

    const checkBoxInput = getByTestId("checkbox");

    expect(checkBoxInput).toBeInTheDocument();
    expect(checkBoxInput).toBeChecked();

    fireEvent.click(checkBoxInput!);
    expect(handleCheckboxMock).toHaveBeenCalledTimes(1);
  });
});
