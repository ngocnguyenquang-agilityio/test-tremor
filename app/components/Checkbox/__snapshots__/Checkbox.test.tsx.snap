// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox Component renders Checkbox component snapshot correctly 1`] = `
<div>
  <input
    checked=""
    class="form-checkbox text-primary h-5 w-5 border-rarely rounded-md checked:bg-[length:90%_90%] border-[1px] checked:bg-[length:80%_80%] border-[1px] checked:bg-[length:50%_50%] border-[1px] opacity-100"
    data-testid="checkbox"
    tabindex="3"
    type="checkbox"
  />
</div>
`;
